librz/core/disasm.c

```diff
@@ -4099,6 +4099,12 @@ static void ds_print_ptr(RzDisasmState *ds, int len, int idx) {
 			}
 		}
 		rz_io_read_at(core->io, refaddr, (ut8 *)msg, len - 1);
+		if (ds->analysis_op.type == RZ_ANALYSIS_OP_TYPE_LOAD) {
+			ut32 referenced_value = ((ut32 *)msg)[0];
+			ds_begin_comment(ds);
+			ds_comment(ds, true, "; refval: 0x%08" PFMT64x, referenced_value);
+		}
+
```

Comment:
probably you want to ensure that the `refptr` value is initialized for load instructions in the arch. also there is a host endianness issue where msg can be little endian as big endian, making this code outputting different values if analyzed on hosts with different endianness.

Reply:
probably you want to add it here: https://github.com/rizinorg/rizin/blob/dev/librz/analysis/p/analysis_xtensa.c#L70

Reply:
for example: https://github.com/rizinorg/rizin/blob/dev/librz/analysis/p/analysis_avr.c#L159

Reply:
Ok, so took a look at `refptr`, and actually setting that to 4 for op `l32r` has a perfect outcome:

`0x4000bf52      l32r  a5, 0x400003f4                       ; [0x400003f4:4]=0xff00`

I updated the PR, to become a much less intrusive change, which also solves much of the concerns with the original.